source/features/infinite-commit-list-scroll.tsx

```diff
@@ -0,0 +1,92 @@
+import select from 'select-dom';
+import React from 'dom-chef';
+import fetchDom from '../libs/fetch-dom';
+import features from '../libs/features';
+
+// Used feature: infinite-scroll as template
+
+// Button from: https://primer.style/css/components/loaders
+const githubLoadingButton = <button className="btn mn-3" disabled><span>Loading</span><span className="AnimatedEllipsis"></span></button>;
+
+let container: HTMLElement;
+let link: HTMLAnchorElement | undefined;
+
+let lastDate: string;
+
+function getLastDate(document_?: HTMLDivElement): string {
+	if (!document_) {
+		document_ = select<HTMLDivElement>('.commits-listing')!;
+	}
+
+	const lastDivInCommitListing = select('.commits-listing > div:last-of-type', document_);
+	return lastDivInCommitListing!.textContent!;
+}
+
+async function appendOlder(): Promise<void> {
+	// Don't load if we don't have the link
+
+	if (!link?.href) {
+		return;
+	}
+
+	// Replace buttons with loading button
+	select('*', container)!.remove();
+	container.append(githubLoadingButton);
```

Comment:
Rename `container` to `pagination`

Use `pagination.replaceWith(githubLoadingButton)`

Later you can use the loading button as insertion point:

```is
githubLoadingButton.before(…)
githubLoadingButton.replaceWith(…)
```

Reply:
The reason I removed the children and placed the loading button inside `container` is it solves 2 problems, 
1. I don't have to `.disconnect()` and `.observe()` with `inView` when I change something
   * since I don't full understand `IntersectionObserver` don't know what to `.observe()` after calling `githubLoadingButton.replaceWith(…)`
1. It's keeps it style of `container`, so I don't have to add style to the loading button, it aligns it's self